transport-native-kqueue/src/main/java/io/netty/channel/kqueue/NativeLongArray.java

```diff
@@ -28,13 +32,27 @@
         if (capacity < 1) {
             throw new IllegalArgumentException("capacity must be >= 1 but was " + capacity);
         }
-        memoryAddress = PlatformDependent.allocateMemory(capacity * SIZEOF_JLONG);
+        memory = Buffer.allocateDirectWithNativeOrder(calculateBufferCapacity(capacity));
+        memoryAddress = Buffer.memoryAddress(memory);
         this.capacity = capacity;
     }
 
+    private static int idx(int index) {
+        return index * SIZEOF_JLONG;
+    }
+
+    private static int calculateBufferCapacity(int capacity) {
+        return capacity * SIZEOF_JLONG;
+    }
+
     void add(long value) {
-        checkSize();
-        PlatformDependent.putLong(memoryOffset(size++), value);
+        reallocIfNeeded();
+        if (PlatformDependent.hasUnsafe()) {
+            PlatformDependent.putLong(memoryOffset(size), value);
+        } else {
+            memory.putLong(idx(size), value);
+        }
+        ++size;
```

Comment:
is this size the number of longs, or the amount of memory?  Below, capacity is compared to it, but thats in units of bytes, not longs.

Reply:
It’s the number of longs stored. I am not confused by your comment tho as capacity is also not bytes but the max number of longs we can hold. Can you point me to the place that is incorrect ?